File: V1java/product/src/main/java/om/product/dao/ProductRepository.java
package om.product.dao;

import module java.base;
import om.product.model.entity.Product;
import org.springframework.data.domain.Example;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ProductRepository extends MongoRepository<Product, String> {
    Collection<Product> findOneByName();

    default Collection<Product> findAll(String skuCode) {
        Product exampleProduct = new Product();
        exampleProduct.setSkuCode(skuCode);
        return findAll(Example.of(exampleProduct));
    }
}

File: omkt/product/src/main/kotlin/om/product/repo/ProductRepository.kt
package om.product.repo

import om.product.entity.Product
import org.springframework.data.mongodb.repository.MongoRepository

interface ProductRepository : MongoRepository<Product,String> {
    fun findBySkuCodeIgnoreCase(skuCode: String?): List<Product?>?

    // Enabling ignoring case
    fun findByDescIgnoreCase(desc: String?): List<Product?>?

    // Enabling static ORDER BY
    fun findBySkuCodeOrderBySkuCodeAsc(skuCode: String?): List<Product?>?
    fun findBySkuCodeOrderBySkuCodeDesc(skuCode: String?): List<Product?>?
}